src/test/java/use_cases/create_new_month/NewMonthUCITest.java

```diff
@@ -0,0 +1,101 @@
+package use_cases.create_new_month;
+
+import entities.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import views.create_new_month.NewMonthP;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Tests the NewMonthUCI class.
+ */
+class NewMonthUCITest {
+    static NewMonthOB outputBoundary;
+    static int monthID1;
+    static int monthID2;
+    static double budgetValue;
+    static MonthlyStorage monthlyStorage1;
+    static SessionStorage sessionStorage;
+
+    @BeforeEach
+    void setUp() {
+        outputBoundary = new NewMonthP();
+        monthID1 = 202305;
+        monthID2 = 202308;
+        budgetValue = 100;
+        monthlyStorage1 = new MonthlyStorage(monthID1,budgetValue);
+        sessionStorage = new SessionStorage();
+        sessionStorage.addMonth(monthlyStorage1);
+    }
+
+    /**
+     * Test case when new MonthlyStorage is created with correct ID.
+     * @throws EntityException when test fails
+     */
+    @Test
+    void createOutputNoRecurExpenseSuccess() throws EntityException {
+        NewMonthIB interactor = new NewMonthUCI(outputBoundary);
+        NewMonthID inputData = new NewMonthID(sessionStorage,monthID2,budgetValue);
+
+        NewMonthOD output = interactor.createOutput(inputData);
+        assertTrue(output.isSuccessful());
+        assertEquals(monthID2, sessionStorage.getMonthlyData(monthID2).getMonthID());
+    }
+
+    /**
+     * Test case when new MonthlyStorage is created with correct ID, and recurring expenses added.
+     * @throws EntityException when test fails
+     */
+    @Test
+    void createOutputRecurExpenseSuccess() throws EntityException {
+        NewMonthIB interactor = new NewMonthUCI(outputBoundary);
+
+        Category other = sessionStorage.getMonthlyData(monthID1).getCategoryData().get(0);
+        Expense recurring = new Expense("Indigo", other, 30.00);
+        sessionStorage.addRecurExpense(recurring);
+
+        NewMonthID inputData = new NewMonthID(sessionStorage,monthID2,budgetValue);
+
+        NewMonthOD output = interactor.createOutput(inputData);
+        assertTrue(output.isSuccessful());
+        assertEquals(monthID2, sessionStorage.getMonthlyData(monthID2).getMonthID());
+        assertEquals(recurring,sessionStorage.getMonthlyData(monthID2).getExpenseData().get(0));
+    }
+
+    /**
+     * Test case when new MonthlyStorage is not added to session because there were duplicate
```

Comment:
For these tests, you should be using assert.throws() instead of just saying that the EntityException will be thrown. You should only be allowing it to be thrown in correct cases (where you know it won't be thrown).

Reply:
The first two test are success cases so I know they won't be thrown. I'm editing the third test case so it won't do that!